blobs.go

```diff
@@ -0,0 +1,64 @@
+package distribution
+
+import (
+	"io"
+	"time"
+
+	"github.com/docker/distribution/digest"
+)
+
+// Blob describes a readable and seekable binary object. Most registry
+// objects, including layers and manifests can be accessible by digest in a
+// BlobService.
+type Blob interface {
+	// NOTE(stevvooe): http.ServeContent requires an efficient implementation
+	// of ReadSeeker.Seek(0, os.SEEK_END).
+	io.ReadSeeker
+	io.ReaderAt
+	io.WriterTo
+	io.Closer
+
+	// Descriptor identifies the layer.
+	Descriptor() Descriptor
+}
```

Comment:
@diogomonica @NathanMcCauley Perhaps, we have signatures hanging off the blob itself (or maybe the tag).